cpp/collections_module/algorithm/algorithm.cpp

```diff
@@ -0,0 +1,30 @@
+#include "algorithm.hpp"
+
+#include <unordered_set>
+#include "mgp.hpp"
+
+void Collections::ContainsAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto &list1{arguments[0].ValueList()};
```

Comment:
You don't need reference here as you are creating value list1. References are  like aliases.